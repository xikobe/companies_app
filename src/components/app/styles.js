import styled from '@emotion/styled';

export const Wrapper = styled.main(({ theme }) => ({
  display: 'flex',
  flexFlow: 'column',

  [theme.mediaQ[2]]: {
    flexFlow: 'row wrap',
    maxWidth: '1400px',
    margin: '2rem auto',
  },
}));

export const ListWrapper = styled.main(() => ({
  flex: 1,
  padding: '15px',
}));
