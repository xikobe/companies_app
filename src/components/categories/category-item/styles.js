import styled from '@emotion/styled';

export const Item = styled.button((props) => ({
  position: 'relative',
  width: '100%',
  textAlign: 'left',
  display: 'block',
  padding: '1rem',
  fontSize: '17px',
  fontWeight: '300',
  color: '#202121',
  cursor: 'pointer',
  background: props.theme.colors.gray,
  borderBottom: '1px solid #E6EAEA',
  textDecoration: 'none',
  transition: 'all 0.3s ease',

  ':hover': {
    backgroundColor: props.theme.colors.white,
    transition: 'background 0.3s ease-in-out',
  },
}));
