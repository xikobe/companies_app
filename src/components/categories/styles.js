import styled from '@emotion/styled';

export const Wrapper = styled.nav(() => ({
  flex: 0,
  padding: '15px',
}));
