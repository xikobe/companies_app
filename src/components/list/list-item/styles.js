import styled from '@emotion/styled';

export const Wrapper = styled.li(({ theme }) => ({
  display: 'flex',
  padding: '1.5rem',
  backgroundColor: theme.colors.white,
  marginBottom: '1.5rem',
  flexFlow: 'column wrap',
  boxShadow: '0 2px 3px 0 #e8ebed, 0 0 3px 0 #e8ebed',
  transition: 'background .2s',

  ':hover': {
    cursor: 'pointer',
    backgroundColor: `${theme.colors.gray}4d`,
  },
}));

export const Info = styled.div(() => ({
  display: 'flex',
  flexFlow: 'row wrap',
  justifyContent: 'space-between',
  marginBottom: '1.5rem',
}));

export const Subscriptions = styled.ul(() => ({
  display: 'flex',
  flexFlow: 'row wrap',
}));

export const Categories = styled.ul(() => ({
  display: 'flex',
  flexFlow: 'row wrap',
}));

export const Category = styled.li(({ theme }) => ({
  color: theme.colors.teal,
  paddingRight: '4px',
  '&:after': {
    content: '" /"',
  },

  '&:last-of-type': {
    paddingRight: '0',
    '&:after': {
      content: '""',
    },
  },
}));

export const Subscription = styled.li`
  display: flex;
  flex-flow: row wrap;
  align-items: baseline;
  margin-right: 1rem;
  position: relative;

  & sup {
    position: absolute;
    top: -.3rem;
  }
`;
