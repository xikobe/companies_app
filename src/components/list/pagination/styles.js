import styled from '@emotion/styled';

export const PagesWrapper = styled.ul(() => ({
  display: 'flex',
  flexFlow: 'row wrap',
  justifyContent: 'center',
  alignItems: 'center',
}));

const paginationButtonRules = (theme) => ({
  color: theme.colors.grayDark,
  backgroundColor: 'transparent',
  fontWeight: 300,
  fontSize: '1rem',
  lineHeight: 1.5,
  padding: '6px 7px',
  cursor: 'pointer',
  border: '1px solid transparent',
  transition: 'border-color .2s ease',

  '&:hover': {
    borderColor: theme.colors.teal,
  },
});

export const PageItem = styled.button(({ isActive, theme }) => ({
  ...paginationButtonRules(theme),
  ...(isActive && {
    color: theme.colors.white,
    backgroundColor: theme.colors.teal,
  }),
}));

export const PrevButton = styled.button(({ theme }) => ({
  ...paginationButtonRules(theme),

  '&:before': {
    content: '"<"',
  },
}));

export const NextButton = styled.button(({ theme }) => ({
  ...paginationButtonRules(theme),

  '&:before': {
    content: '">"',
  },
}));
