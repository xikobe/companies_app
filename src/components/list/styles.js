import styled from '@emotion/styled';

export const Wrapper = styled.ul();

export const PaginationWrapper = styled.div();
