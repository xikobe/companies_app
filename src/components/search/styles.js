import styled from '@emotion/styled';

export const Input = styled.input(({ theme }) => ({
  width: '100%',
  fontSize: '1rem',
  fontWeight: '300',
  padding: '1rem',
  transition: 'all .2s ease',
  border: `1px solid ${theme.colors.gray}`,
  marginBottom: '2rem',

  ':hover, :focus': {
    cursor: 'auto',
    border: `1px solid ${theme.colors.teal}`,
  },
}));
