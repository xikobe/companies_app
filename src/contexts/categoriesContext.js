import React, {
  createContext, useContext, useState, useCallback,
} from 'react';
import PropTypes from 'prop-types';

const CategoriesContext = createContext();

export const useCategoriesContext = () => useContext(CategoriesContext);

const CategoriesProvider = ({ children }) => {
  const [activeCategories, setActiveCategories] = useState([]);

  const toggleActiveCategory = useCallback((category) => {
    if (activeCategories.includes(category)) {
      return setActiveCategories(activeCategories.filter((cat) => cat !== category));
    }

    return setActiveCategories([...activeCategories, category]);
  }, [activeCategories]);

  const isCategoryActive = useCallback((category) => activeCategories.includes(category), [activeCategories]);

  const clearCategories = () => setActiveCategories([]);

  return (
    <CategoriesContext.Provider value={{
      toggleActiveCategory, isCategoryActive, activeCategories, clearCategories,
    }}
    >
      { children }
    </CategoriesContext.Provider>
  );
};

CategoriesProvider.propTypes = {
  children: PropTypes.node,
};

export default CategoriesProvider;
