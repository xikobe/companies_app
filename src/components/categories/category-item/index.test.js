import React from 'react';
import { shallow } from 'enzyme';
import CategoryItem from './index';
import { useCategoriesContext } from '../../../contexts/categoriesContext';
import { Item } from './styles';

jest.mock('../../../contexts/categoriesContext');

const mockToggle = jest.fn();
useCategoriesContext.mockImplementation(() => ({
  toggleActiveCategory: mockToggle,
  isCategoryActive: () => false,
}));

describe('CategoryItem', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CategoryItem name="Foo" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Item).prop('isActive')).toBe(false);
  });

  it('should call toggleActiveCategory on click', () => {
    const wrapper = shallow(<CategoryItem name="Foo" />);

    wrapper.find(Item).simulate('click');

    expect(mockToggle).toHaveBeenCalledWith('Foo');
  });
});
