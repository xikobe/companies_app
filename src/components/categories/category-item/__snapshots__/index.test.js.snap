// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem renders correctly 1`] = `
<li>
  <Styled(button)
    isActive={false}
    onClick={[Function]}
  >
    Foo
  </Styled(button)>
</li>
`;
